Algorithm/QCAlgorithm.Framework.cs

```diff
@@ -382,6 +384,26 @@ public void EmitInsights(params Insight[] insights)
                 return;
             }
 
+            // Verify insights are valid
+            var validInsights = new List<Insight>();
+            foreach(var insight in insights){
+                if (Securities[insight.Symbol].IsDelisted)
+                {
+                    if (!_isEmitDelistedInsightWarningSent)
+                    {
+                        Error($"QCAlgorithm.EmitInsights(): Warning: cannot emit insights for delisted securities, these will be discarded");
```

Comment:
Cool! 👍 
We are missing to cover the insight emitted by the alpha too, see `public void OnFrameworkData(Slice slice)`. 
Suggest we create a method that performs this loop and `InitializeInsightFields` at the same time, returning an array, so we don't iterate twice over the insights. `OnInsightsGenerated` should also take an array of insights. And we call this new method in both locations.
We should add a regression test covering these two cases too